src/App.tsx

```diff
@@ -15,49 +16,128 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((firstGood, secondGood) => (
+        firstGood.localeCompare(secondGood)
+      ));
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((firstGood, secondGood) => (
+        secondGood.length - firstGood.length
+      ));
```

Comment:
Should be 'firstGood.length - secondGood.length'.... your indicator of active Reverse button works wrong, change it there - 116 row.

Reply:
fixed it